src/TensorFlowNET.Core/Buffers/SafeBufferHandle.cs

```diff
@@ -0,0 +1,39 @@
+﻿/*****************************************************************************
+   Copyright 2018 The TensorFlow.NET Authors. All Rights Reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+******************************************************************************/
+
+using System;
+using Tensorflow.Util;
+
+namespace Tensorflow
+{
+    public sealed class SafeBufferHandle : SafeTensorflowHandle
+    {
+        public SafeBufferHandle()
```

Comment:
Should we make this constructor private? Becasue when user call `new SafeBufferHandle()` it doesn't impact anything without `handle`.

Or is it better that moving `c_api.TF_NewBuffer()` into constructor?

Reply:
> Should we make this constructor private?

Based on `SafeFileHandle`, it looks like we can make this constructor private. Can you file a follow-up issue for this one since there are a few other cases already merged I'd like to verify as soon as the tests are passing again?

Reply:
Actually I just submitted a second PR for all the other cases (#566).

Reply:
➡️ Updated `SafeBufferHandle` in this pull request, and the remaining constructors in #566.